parser/parser_test.go

```diff
@@ -342,12 +349,24 @@ world`},
 				Left:  &IdentifierNode{Value: "foo"},
 				Right: &StringNode{Value: "foo"}},
 		},
+		{
+			`startsWith(foo,  "foo")`,
+			&BinaryNode{Operator: "startsWith",
+				Left:  &IdentifierNode{Value: "foo"},
+				Right: &StringNode{Value: "foo"}},
+		},
 		{
 			`foo endsWith "foo"`,
 			&BinaryNode{Operator: "endsWith",
 				Left:  &IdentifierNode{Value: "foo"},
 				Right: &StringNode{Value: "foo"}},
 		},
+		{
+			`endsWith(foo,  "foo")`,
```

Comment:
Let's also add test for postfix/prefix and other combinations as:
```
endsWith() endsWith endsWith()
endsWith().Field
endsWith()[0]
-+endsWith()
endsWith()+endsWith()
```
To see how parser whould behaive in complex scenarios?

Reply:
added more tests cases as per your feedback